### DISCUSSION TITLE
esc to quit

### DISCUSSION THREAD

**Discussion by bennypowers** [jesseduffield/lazygit#1966] [Q&A] [answered]
Lately I've been using *lazygit* in *neovim* via *toggleterm* to do a lot of my git stuff. `<leader>g` opens a floating terminal and runs lazygit. The floating terminal automatically closes when the process exits cleanly with a 0 exit code.

What I want to do is close the floating terminal when I press escape at the 'top level' of lazygit's ui. So for example if, starting in normal mode in neovim, I did:

```
<leader>g -- launch lazygit in floating term
l         -- switch to branches pane
enter     -- drill into selected branch
esc       -- drill up back into branches pane
esc       -- 👈 
```
At the 👈 , I'm back to the top-level of lazygit's ui heirarchy. I'd like that `<esc>` keypress to exit lazygit, so I can get back to my vim buffer. Glancing through the customCommands examples, I didn't see a ready example of killing the current lazygit process. Is this something that can be configured, or would it require a change in lazygit?

**Comment by bennypowers** [comment, marked as answer]
found the top-level config flag:
```yaml
﻿﻿﻿quitOnTopLevelReturn: true
```

**Reply by medv** [reply]
Awesome, had the same gripe a year later. Thanks!

**Reply by erlangparasu** [reply]
where i need to put that option?

**Reply by mark2185** [reply]
In the path specified in the [documentation](https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md).

**Reply by easystartup-io** [reply]
This works. thanks so much

**Reply by beipym** [reply]
didnt work for me

should i put it under gui?
```
gui:
     ﻿﻿quitOnTopLevelReturn: true
```
or do i need to somehow notify the lazygit the settings have changed? ( eg: source the config ) if yes, how?

**Reply by stefanhaller** [reply]
@beipym No, not under gui, it's top-level exactly as written above.

If this doesn't work for you, my guess would be that you are not using the right location for the config file. Type `1 e` from within lazygit to find out exactly which config file lazygit is using.

**Reply by beipym** [reply]
@stefanhaller 
its the correct file
when i use `1 e` i can even see the ` quitOnTopLevelReturn: true` but its not working, and it is top-level just like it should be 

anyway thanks for helping 

( after tinkering a lot, i end up using raw git in terminal, and im happy ! )
